core/txpool/blobpool/blobpool.go

```diff
@@ -871,6 +885,70 @@ func (p *BlobPool) Reset(oldHead, newHead *types.Header) {
 	basefeeGauge.Update(int64(basefee.Uint64()))
 	blobfeeGauge.Update(int64(blobfee.Uint64()))
 	p.updateStorageMetrics()
+
+	// Perform the conversion logic at the fork boundary
+	if !p.chain.Config().IsOsaka(oldHead.Number, oldHead.Time) && p.chain.Config().IsOsaka(newHead.Number, newHead.Time) {
+		// Evicts all transactions from the pool. The pool should, at this stage,
+		// contain only legacy blob transactions. Evicting any new-format blob
+		// transactions is also safe, as they will be re-injected later.
+		txs := make(map[common.Address]map[uint64]uint64)
+		for p.stored > 0 {
+			sender, nonce, id := p.drop(false)
+			if txs[sender] == nil {
+				txs[sender] = make(map[uint64]uint64)
+			}
+			txs[sender][nonce] = id
+		}
+		// Initiate the background conversion thread, the mutex is not required
+		// and won't block any pool operation.
+		go func() {
+			var (
+				start   = time.Now()
+				success int
+				fail    int
+			)
+			for addr, list := range txs {
+				// Transactions evicted from the pool must be contiguous, if in any case,
+				// the transactions are gapped with each other, they will be discarded.
+				nonces := slices.Collect(maps.Keys(list))
+				slices.Sort(nonces)
+
+				// Convert and insert the txs in order
+				for _, nonce := range nonces {
+					id := list[nonce]
+					blob, err := p.store.Get(id)
+					if err != nil {
+						fail++
+						continue
+					}
+					// Remove the blob transaction regardless of conversion succeeds or not
+					if err := p.store.Delete(id); err != nil {
```

Comment:
In case of the happy path, wouldn't it be better to remove just before re-adding the converted version? Not sure it has any significance, since the v0 version is useless by the time we start converting, just thinking aloud.